### PR TITLE
feat(auth): use minimal scopes for CC and CW

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
The Toolkit requests permissions for CodeWhisperer and CodeCatalyst even if the user doesn't want to use both

## Solution
Request the minimum permissions required to use a feature

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]

**Comment by JadenSimon** [comment]
`crypto.randomUUID` isn't in min ver I guess

File: src/credentials/secondaryAuth.ts [hunk 1]
```diff
@@ -58,10 +64,27 @@ async function promptUseNewConnection(newConn: Connection, oldConn: Connection,
...
 }
 
+async function promptForRescope(conn: SsoConnection, toolLabel: string) {
+    const message = localize(
+        'aws.auth.rescopeConnection.message',
+        '{0} requires access to your {1} connection to continue. Proceed to login to grant {0} access?',
+        toolLabel,
+        conn.startUrl === builderIdStartUrl ? localizedText.builderId() : localizedText.iamIdentityCenter
+    )
+    const resp = await vscode.window.showInformationMessage(message, { modal: true }, localizedText.proceed)
```
Comment (justinmk3) [inline src/credentials/secondaryAuth.ts:74]:
```suggestion
    const resp = await vscode.window.showInformationMessage(message, { modal: true }, localizedText.continue)
```

File: src/credentials/secondaryAuth.ts [hunk 2]
```diff
@@ -58,10 +64,27 @@ async function promptUseNewConnection(newConn: Connection, oldConn: Connection,
     return resp
 }
 
+async function promptForRescope(conn: SsoConnection, toolLabel: string) {
+    const message = localize(
+        'aws.auth.rescopeConnection.message',
+        '{0} requires access to your {1} connection to continue. Proceed to login to grant {0} access?',
```
Comment (justinmk3) [inline src/credentials/secondaryAuth.ts:70]:
```suggestion
        '{0} requires access to your {1} connection. Continue to login to grant {0} access?',
```

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**